File: Study-Code/Some_Function_C++/Search/Binary_search.cpp
#include <iostream>
using namespace std;

int binary_search(int arr[],int n,int target) 
{
    int left = 0;
    int right = n - 1;
    while(left<=right)
    {
        int mid = (left + right) / 2;
        if(arr[mid] < target)
            left = mid + 1;
        else if(arr[mid] == target)
            return mid;
        else 
            right = mid - 1;
    }
    return -1;
}

int main()
{
    int n, target; 
    cin >> n >> target;
    int arr[n];
    
    for(int i = 0; i < n; i++) 
    {
        cin >> arr[i];
    }
    
    int res = binary_search(arr,n,target);
    cout << res;

    return 0;
}

File: Study-Code/Some_Function_C#/Search_Algorithm/Binary_Search.cs
using System;

namespace Practice_C_Sharp
{
    internal class Program
    {
        static int Binary_Search(int[] arr, int size, int target)
        {
            int left = 0;
            int right = size - 1;

            while(left <= right)
            {
                int mid = (left + right) / 2;

                if (arr[mid] < target)
                {
                    left = mid + 1;
                }
                else if (arr[mid] == target)
                {
                    return mid;
                }
                else
                {
                    right = mid - 1;
                }
            }
            return -1;
        }
        public static void Main(string[] args)
        {
            int size = Convert.ToInt32(Console.ReadLine());
            int target = Convert.ToInt32(Console.ReadLine());
            int[] array = new int[size];

            for(int  i = 0; i < size; i++)
            {
                array[i] = Convert.ToInt32(Console.ReadLine());
            }

            int result = Binary_Search(array, size, target);
            Console.Write($"Position of target is: {result}");

        }
    }
}
